confgenerator/otel/modular.go

```diff
@@ -143,13 +146,20 @@ func (c ModularConfig) Generate() (string, error) {
 		"service":    service,
 	}
 
-	resourceDetectionProcessors := map[ResourceDetectionMode]Component{
+	resourceProcessors := map[ResourceDetectionMode]Component{
 		Override: GCPResourceDetector(true),
 		Upsert:   GCPResourceDetector(false),
 	}
-	resourceDetectionProcessorNames := map[ResourceDetectionMode]string{
-		Override: resourceDetectionProcessors[Override].name("_global_0"),
-		Upsert:   resourceDetectionProcessors[Upsert].name("_global_1"),
+
+	if pl.ResourceOverride != nil {
+		resourceProcessors = map[ResourceDetectionMode]Component{
+			Override: ResourceTransform(pl.ResourceOverride.OTelResourceAttributes()),
+			Upsert:   ResourceTransform(pl.ResourceOverride.OTelResourceAttributes()),
```

Comment:
It looks like you're still not handling the two cases differently. That said, it also looks like I've named these enum cases poorly. The "override" case should be replacing the existing resource attributes (which for the `resource` processor means `"action": "upsert"`), and the other case should be only setting the resource if it's not already set (which for the `resource` processor means `"action": "insert"`). See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/attributesprocessor/README.md

So change `ResourceTransform` to also take a bool parameter like `GCPResourceDetector` that changes what it sets `action` to, and if you want rename `Upsert` to `SetIfMissing` or something like that to make it clearer.

Reply:
Done.